internal/output_workflow/findings_model_tools.go

```diff
@@ -0,0 +1,236 @@
+				renderEmptyData: fileWriter.WriteEmptyContent,
+			}
+		}
+	}
+
+	return writerMap
+}
+
+func getDefaultWriter(config configuration.Configuration, outputDestination OutputDestination) *WriterEntry {
+	writer := &WriterEntry{
+		writer: &newLineCloser{
+			writer: outputDestination.GetWriter(),
+		},
+		mimeType:        DEFAULT_MIME_TYPE,
+		templates:       presenters.DefaultTemplateFiles,
+		renderEmptyData: true,
+	}
+
+	if config.IsSet(OUTPUT_CONFIG_TEMPLATE_FILE) {
+		writer.templates = []string{config.GetString(OUTPUT_CONFIG_TEMPLATE_FILE)}
+	}
+
+	return writer
+}
+
+func useRendererWithUnifiedModel(name string, wEntry *WriterEntry, projectResults []*presenters.UnifiedProjectResult, invocation workflow.InvocationContext) {
+	debugLogger := invocation.GetEnhancedLogger()
+
+	if !wEntry.renderEmptyData && getTotalNumberOfUnifiedFindings(projectResults) == 0 {
+		debugLogger.Info().Msgf("[%s] The input is empty, skipping rendering!", name)
+		return
+	}
+
+	debugLogger.Info().Msgf("[%s] Creating findings model renderer", name)
+
+	defer func() {
+		closeErr := wEntry.writer.Close()
+		if closeErr != nil {
+			debugLogger.Err(closeErr).Msgf("[%s] Error while closing writer.", name)
+		}
+	}()
+
+	config := invocation.GetConfiguration()
+	renderer := presenters.NewUnifiedFindingsRenderer(
+		projectResults,
+		config,
+		wEntry.writer,
+		presenters.WithUnifiedRuntimeInfo(invocation.GetRuntimeInfo()),
+	)
+
+	debugLogger.Info().Msgf("[%s] Rendering %s with %s", name, wEntry.mimeType, wEntry.templates)
+	err := renderer.RenderTemplate(wEntry.templates, wEntry.mimeType)
+	if err != nil {
+		debugLogger.Warn().Err(err).Msgf("[%s] Failed to render local finding", name)
+		return
+	}
+
+	debugLogger.Info().Msgf("[%s] Rendering done", name)
+}
+
+func HandleContentTypeUnifiedModel(input []workflow.Data, invocation workflow.InvocationContext, outputDestination OutputDestination) ([]workflow.Data, error) {
+	var err error
+	debugLogger := invocation.GetEnhancedLogger()
+	config := invocation.GetConfiguration()
+
+	projectResults, remainingData := getUnifiedProjectResults(input, debugLogger)
+	if len(projectResults) == 0 {
+		debugLogger.Info().Msg("No complete projects with findings and summary to process")
+		return remainingData, nil
+	}
+
+	threadCount := max(int64(config.GetInt(configuration.MAX_THREADS)), 1)
+	debugLogger.Info().Msgf("Thread count: %d", threadCount)
+
+	writerMap := getWritersToUse(config, outputDestination)
+
+	// iterate over all writers and render for each of them
+	ctx := context.Background()
+	availableThreads := semaphore.NewWeighted(threadCount)
+	for k, v := range writerMap {
+		err = availableThreads.Acquire(ctx, 1)
+		if err != nil {
+			debugLogger.Err(err).Msgf("[%s] Failed to acquire threading lock. Cancel rendering.", k)
+			break
+		}
+
+		go func(name string, writer *WriterEntry) {
+			defer availableThreads.Release(1)
+			useRendererWithUnifiedModel(name, writer, projectResults, invocation)
```

Comment:
Could / should we collect errors here and bubble them up through a channel? Something to consider for a followup.

I could see a case where a failure to write results should cause an exit code 2, if those results are important to record for compliance etc.